app/src/main/java/com/example/myapplication/ui/LoginFragment.kt

```diff
@@ -75,16 +77,38 @@ class LoginFragment : Fragment() {
     }
 
     private fun userResponseObserver() {
-        loginViewModel.userResponseIsSuccessful.observe(viewLifecycleOwner) {
-            if (it == true) {
-                Toast.makeText(context, SUCCESS, Toast.LENGTH_SHORT)
-                    .show()
-                this.findNavController().navigate(
-                    LoginFragmentDirections.actionLoginFragmentToHomePageFragment()
-                )
-            } else {
-                Toast.makeText(context, FAILURE, Toast.LENGTH_SHORT)
-                    .show()
+        loginViewModel.userResponse.observe(viewLifecycleOwner) { responseState ->
+            binding.progressBar.visibility = View.GONE
+            when (responseState) {
+                NetworkResponseState.SUCCESS -> {
+                    this.findNavController().navigate(
+                        LoginFragmentDirections.actionLoginFragmentToHomePageFragment()
+                    )
+                }
+                NetworkResponseState.INVALID_CREDENTIALS -> {
+                    Toast.makeText(
+                        context,
+                        CREDENTIALS,
+                        Toast.LENGTH_SHORT
+                    )
+                        .show()
+                }
+                NetworkResponseState.NO_INTERNET_CONNECTION -> {
+                    Toast.makeText(
+                        context,
+                        INTERNET,
+                        Toast.LENGTH_SHORT
+                    )
+                        .show()
+                }
+                NetworkResponseState.EXCEPTION -> {
+                    Toast.makeText(
+                        context,
+                        INTERNET,
```

Comment:
Just a silly comment. This is a generic error, so we should show a generic message. What about "Something has failed, try again later"? 
Also, you could use the `else` branch here.